nep-xx.mediawiki

```diff
@@ -0,0 +1,103 @@
+|                                                              |      | "safe": false,                                               |
+|                                                              |      | "parameters": [                                              |
+|                                                              |      | {                                                            |
+|                                                              |      | "name": "data",                                              |
+|                                                              |      | "type": "Any"                                                |
+|                                                              |      | },                                                           |
+|                                                              |      | {                                                            |
+|                                                              |      | "name": "update",                                            |
+|                                                              |      | "type": "Boolean"                                            |
+|                                                              |      | }                                                            |
+|                                                              |      | ],                                                           |
+|                                                              |      | "returntype": "Void"                                         |
+|                                                              |      | }                                                            |
+|                                                              |      | </pre>                                                       |
+|                                                              |      |                                                              |
+|                                                              |      | This is an optional function which will be automatically executed when contract first deployed. |
+|                                                              |      |                                                              |
+|                                                              |      | The parameter <code>data</code> can be any type of supported parameters for additional message purpose. |
+|                                                              |      |                                                              |
+|                                                              |      | <code>update</code> returns if it's the initial deployment of this contract. The function SHOULD check whether <code>update</code> returns true. If not, continue execution. If true, return. |
```

Comment:
> The function SHOULD check whether <code>update</code> returns true. If not, continue execution. If true, return.

I don't think this is correct advice. It seems perfectly reasonable to me to do something like this (pseudo)

```
deploy(data, update) {
   if !update {
      // initial deploy, put default values into storage
   } else {
      // reset part of storage to default
   }
}
```

Reply:
In fact, either `initial deploy` or `update` will send notification, no need to waste storage.
https://github.com/neo-project/neo/blob/5f4c14e3801c691ceccb5705c634e9e55dd3bba1/src/Neo/SmartContract/Native/ContractManagement.cs#L99

Reply:
I'm aware of the existing notification, but that's not the issue I'm trying to point out. The way I'm understanding the text is that it suggests to do the following
```
_deploy(data, update) {
   if update { return }
   // else continue execution of remaining statements
}
```
if that's the case then the whole function should not have a need to have an `update` parameter to begin with. Since it does have this param, it is reasonable that a contract might do something different when updated vs when first deployed. The suggestion would therefore be to remove this text
> The function SHOULD check whether update returns true. If not, continue execution. If true, return.

Reply:
Done